src/components/MapArea.js

```diff
@@ -80,6 +67,22 @@ export default function MapArea() {
     }
   };
 
+  const clearDataFromMap = (map) => {
+    map.current.removeLayer("trailData");
+    map.current.removeSource("trailData");
+  };
+
+  const recenterMap = (map, selectedTrail) => {
+    const { bbox } = selectedTrail.features[selectedTrail.features.length - 1].geometry;
+    map.current.fitBounds(
+      [[bbox[2], bbox[1]], [bbox[0], bbox[3]]],
```

Comment:
it might be worth using array destructuring here to label each of these coordinates, so that it is obvious which one is which direction (north, east, west, south)
```
const [west, south, east ,north] = bbox
```
^ not sure that is the correct order